src/wp-includes/class-wp-script-modules.php

```diff
@@ -30,6 +30,22 @@ class WP_Script_Modules {
 	 */
 	private $enqueued_before_registered = array();
 
+	/**
+	 * Holds script module identifiers that have been marked for inclusion in the import map.
+	 *
+	 * A script module that appears here should be include in the import map regardless of
+	 * whether it is a dependency of another script module.
+	 *
+	 * The values in this array are always `null`. The presence of a Module IDs
+	 * as an array key marks the script module for inclusion in the import map.
+	 * Different values are reserved for possible future use.
```

Comment:
With this all being `private`, this is a premature optimization. We can just use array values for now and once we need more data it can be easily added without back compat concerns.

Reply:
It was not intended as an optimization, but as an array or unique values or set-like behavior.

Since you've mentioned it, uniqueness is likely irrelevant and the array_unique call when printing the import map can also be dropped.